### DISCUSSION TITLE
Lammps with multi GPU

### DISCUSSION THREAD

**Discussion by Lewis-YL** [deepmodeling/deepmd-kit#1812] [Q&A]
Hello,

When I tried lammps with either:
mpirun -np 2 lmp -i in.lammps
or 
horovodrun -np 2 lmp -i in.lammps

It creates 2 job tasks on 2 GPU and it runs at the same speed as with 1 GPU.
Here is a screen shot.

![image](https://user-images.githubusercontent.com/38740371/177218504-e68f8ffc-d16e-4d90-b627-35f197febc74.png)

Do you know how to run a lammps on multi GPU?

**Comment by njzjz** [comment]
This should be expected if your GPU utilization is low. You may need to know the working mode of GPUs...

**Reply by Lewis-YL** [reply]
I am using 2 RTX3080Ti, deepmd v2.1.3 with cuda 11.6

Can you tell more about how to run lammps with multi GPU?

I think the speed should be 2X or at least 1.5X.

However, the program just created 2 same jobs on 2 GPUs and did not increase the speed at all.

When the GPU was working, the utilization was always 90%. The mode was always P2 under working and P8 idle.
Is this normal?

**Reply by njzjz** [reply]
How many atoms are there in the system? On your screenshot, I see that only ~400 MB of GPU memory is used by a single `lmp` process...

**Reply by njzjz** [reply]
Note that utilization in `nvidia-smi` doesn't mean how many cores are used actively. It's "percent of time over the past sample period during which one or more kernels was executing on the GPU".

**Reply by Lewis-YL** [reply]
I uploaded the new screenshot of when running the command:

horovodrun -np 2 lmp -i in.lammps

I have 104 atoms in my system.

You can see in the screenshot that there are 2 same runs on the 2 GPUs (id: 77634, 77635).

**Reply by njzjz** [reply]
The system of 104 atoms are too small to utilize all GPU cores.

You may firstly test on a single GPU card. Increase the number of atoms and see how the timing scales.
